index.html

```diff
@@ -18852,4 +18929,495 @@
+</script>
+
+<!-- Similarity comparitors and utils -->
+<script>
+	// a list of commonly used words that have little meaning and can be excluded
+	// from analysis.
+	const stopwords = [
+		'about', 'above', 'after', 'again', 'all', 'also', 'am', 'an', 'and', 'another',
+		'any', 'are', 'as', 'at', 'be', 'because', 'been', 'before', 'being', 'below',
+		'between', 'both', 'but', 'by', 'came', 'can', 'cannot', 'come', 'could', 'did',
+		'do', 'does', 'doing', 'during', 'each', 'few', 'for', 'from', 'further', 'get',
+		'got', 'has', 'had', 'he', 'have', 'her', 'here', 'him', 'himself', 'his', 'how',
+		'if', 'in', 'into', 'is', 'it', 'its', 'itself', 'like', 'make', 'many', 'me',
+		'might', 'more', 'most', 'much', 'must', 'my', 'myself', 'never', 'now', 'of', 'on',
+		'only', 'or', 'other', 'our', 'ours', 'ourselves', 'out', 'over', 'own',
+		'said', 'same', 'see', 'she', 'should', 'since', 'so', 'some', 'still', 'such', 'take', 'than',
+		'that', 'the', 'their', 'theirs', 'them', 'themselves', 'then', 'there', 'these', 'they',
+		'this', 'those', 'through', 'to', 'too', 'under', 'until', 'up', 'very', 'was',
+		'way', 'we', 'well', 'were', 'what', 'where', 'when', 'which', 'while', 'who',
+		'whom', 'with', 'would', 'why', 'you', 'your', 'yours', 'yourself',
+		'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
+		'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '$', '1',
+		'2', '3', '4', '5', '6', '7', '8', '9', '0', '_'
+	]
+
+
+
+	/**
+	 * Basic structure of a Similarity Comparitor - This does not work by itself
+	 */
+	class SimilarityComparitor {
+		/**
+		 * Determines if you split based on paragraphs or instruct / user turns
+		 */
+		_splitAsParagraphs
+		/**
+		 * Amount to trim from the history which is the context
+		 */
+		_amountToTrimForContext
+		/**
+		 * Max length of search term
+		 */
+		_maxLengthOfSearchString
+		/**
+		 * Max length per chunk
+		 */
+		_chunkSplitSize
+		/**
+		 * Overlap per chunk
+		 */
+		_chunkOverlap
+
+		constructor(amountToTrimForContext, maxLengthOfSearchString = 400, chunkSplitSize = 600, chunkOverlap = 200) {
```

Comment:
This can sometimes make a difference if we want to allow a user to change it?

Reply:
Done